contrib/config/source/kv_store_xds_delegate.cc

```diff
@@ -0,0 +1,127 @@
+#include "contrib/config/source/kv_store_xds_delegate.h"
+
+#include "envoy/registry/registry.h"
+#include "envoy/service/discovery/v3/discovery.pb.h"
+
+#include "source/common/common/logger.h"
+#include "source/common/config/utility.h"
+#include "source/common/protobuf/utility.h"
+
+#include "absl/strings/match.h"
+#include "absl/strings/str_cat.h"
+#include "absl/types/optional.h"
+#include "contrib/envoy/extensions/config/v3alpha/kv_store_xds_delegate_config.pb.h"
+#include "contrib/envoy/extensions/config/v3alpha/kv_store_xds_delegate_config.pb.validate.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Config {
+namespace {
+
+using ::Envoy::Config::XdsSourceId;
+using ::envoy::extensions::config::v3alpha::KeyValueStoreXdsDelegateConfig;
+
+// TODO(abeyad): Add a per-Envoy instance prefix to the storage key, to allow xDS resources to be
+// persisted per-running-Envoy when Envoy is embedded as a library (e.g. Envoy Mobile).
```

Comment:
Hm, this is an interesting one.  I wonder if we can get away with a winner-takes-all solution here - I'm interested in @RyanTheOptimist thoughts

Reply:
Hm. I'm not quite sure what is meant by winner-takes-all. Is the thinking that multiple Envoy instances would share the same KV store and whoever gets there first owns it? Or something else?

Reply:
Actually, I think we can ensure different Envoy engines can have different key prefixes to avoid collisions by setting different keys in https://github.com/envoyproxy/envoy-mobile/blob/main/library/common/extensions/key_value/platform/platform.proto#L11

Reply:
Ok in that case do we need a per-instance prefix?

Reply:
We don't, I pushed a commit removing this comment.